File: python-for-cote/3/3-4.py
n, k = map(int, input().split())
result = 0

#n이 k 이상이라면 k 로 계속 나누기
while True:
    # (N == K로 나누어 떨어지는 수가 될 때 까지 1씩 빼기
    target = (n // k) * k
    result += (n - target)
    if result == 4:
        print("target :", target)
        print("222 n :", n)

    print("1 result : ", result)
    n = target
    # n이 k보다 작을 때 (더 이상 나눌 수 없을 때) 반복문 탈출
    if n < k:
        break
    #k로 나누기 
    result += 1
    print("result : ", result)
    n //= k

#마지막으로 남은 수에 대하여 1씩 빼기
print("(n - 1): ",(n - 1))
result += (n - 1)
print(result)

File: python-for-cote/3/3-4.cpp
#include <bits/stdc++.h>

using namespace std;

int n, k;
int result;

int main(){
    //N, K 를 공백을 기준으로 구분하여 입력 받기
    cin >> n >> k;
    
    while(true){
        //N이 K로 나누어 떵러지는 수가 될 때까지만 1씩 빼기;
        int target = (n / k)  * k;

        result += (n - target);
        n = target;
        // n 이 k 보다 작을 때 (더 이상 나눌 수 없을 때) 
        if (n < k) break;
        //k 로 나누기 
        result += 1;
        n /= k;        
    }

    // 마지막으로 남은 수에 대해서 1씩 빼기 
    result += (n - 1);
}